lib/internal/process/modules.js

```diff
@@ -31,15 +45,65 @@ exports.ESMLoader = undefined;
 exports.setup = function() {
   setInitializeImportMetaObjectCallback(initializeImportMetaObject);
 
-  let ESMLoader = new Loader();
+  const ESMLoader = new Loader();
```

Comment:
Note that this was a `let` because a separate loader was used to load the hook loaders than to load the main in order to support that any dependencies of the loaders themselves can support hooking in the main app.

Reply:
this is a regression, nice catch. should we give each loader its own module map space, or just the loaders vs runtime?

Reply:
i've split this into 2 distinct loaders. I did notice a few odd things right now outside of this PR:

1. it looks like `import()` always uses the bootstrap loader if a custom loader exists : https://github.com/nodejs/node/blob/3a191229418dcc0e21956847993b1702c88a923b/lib/internal/process/modules.js#L34-L51

2. it looks like the bootstrap loader and the runtime loader are sharing `require.cache` so the isolation doesn't fully hold for CJS stuff.